project-koku/koku
# Large data automation makefile

## Description

Additions to makefile to drive large data generation. At the current time, this requires nise repo to also be checked out into a sibling directory to koku.

## Commands

* `create-large-ocp-provider-config-file` (params: generator_config_file, generator_template_file, output_file_name, generator_flags) All but generator_flags are required. This command, for now, requires that the nise repo be checked out in a sibling directory to koku. This will generate .yml files that nise can use to create data.
* `create-large-ocp-provider-testing-files` (param: nise_config_dir) Specify a directory of configuration files and this will call nise in a loop to process each `.yml` file in the directory.
* `import-large-ocp-provider-testing-costmodel` will import a static cost model using the provider uuid from a `make create-large-ocp-provider-testing-files` run
* `import-large-ocp-provider-testing-data` calls the cost management download endpoint to ingest the ocp test data
* `large-ocp-provider-testing` (param: nise_config_dir) calls `make create-large-ocp-provider-testing-files`, `make import-large-ocp-provider-testing-costmodel`, `make import-large-ocp-provider-testing-data`
* `purge-large-testing-ocp-files` remove testing files specifically for provider named **large_ocp_1**
* `purge-all-testing-ocp-files` remove all testing ocp data files
* `find-large-testing-provider-uuid` calls endpoint to get the provider uuid for **large_ocp_1**

## Testing

1. Have koku and nise checked out to sibling directories.
2. Make a copy of the config file in nise/utility/generator_config.yml and change the integer values to something very small
3. Make a new dir somewhere.
3. cd to the koku repo
4. `make create-large-ocp-provider-config-file generator_config_file=<your config file here> generator_template_file=../nise/utility/ocp_static_data.yml.j2 output_file_name=<my_new_dir>/my_test.yml`
5. Examine the output file. The values should match your set max values in the generator config
6. `make large-opc-provider-testing nise_config_dir=<my_new_dir>`
7. This should generate nise data, create the provider, load the cost model, and load the ocp data into the database

## Note

This was not designed to run under local OpenShift
